exit with non-zero code when timer cancelled

nathanielknight opened the pull request:
This commit is the minimal possible implementation of #2.

This *might* break backwards compatibility if someone is depending on the current behaviour (exit with `0` even when cancelled). A flag to enable the new behaviour would prevent that at the expense of having a more complicated interface; I'm happy to put that together if it would be preferable.

nathanielknight committed:
exit with non-zero code when timer cancelled

elpnt commented:
`cargo clippy -- -D warning` says mutable vec references `&mut Vec<String>` in line 53, 67 (and also 42) of `src/timer.rs` should be mutable slices `&mut [String]`.

I think fixing these 3 lines would pass the CI.

nathanielknight committed:
Replace Vec with Slice to avoid creating new objects

This commit replaces `&mut Vec<String>` with `&mut [String]` in the
signatures of some of the functions in `src/timer.rs`, which lets us
avoid allocating a new object unnecessarily (per this  Clippy lint:
https://news.ycombinator.com/item?id=32430182).

elpnt commented:
thanks!

close #2
